### ISSUE TITLE
Anaconda fails to include all needed DLLs with 2.5 (was The specified module could not be found: PIL)

### ISSUE THREAD

**Issue by iegorval** [Nuitka/Nuitka#3261]
Hello,

I have encountered the issue that seems to be very similar to https://github.com/Nuitka/Nuitka/issues/2332 but concerns `PIL` package. I am not sure if the issue is in my setup or in Nuitka.

I run my project with `pixi` (https://prefix.dev/), hence the modified command below to get the actual version of `nuitka` used for the build. I am unfortunately not sure at the moment how to install with `pixi` from source but if it is required to understand the source of issue, I can somehow try to do it or temporary move the project to different tool. Important bit here is probably that `pixi` uses Anaconda Python.
          
  > pixi run python -m nuitka --version

```
2.5.6                                                                                                                                                                                                                       
Commercial: None                                                                                                                                                                                                     
Python: 3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:12:32) [MSC v.1929 64 bit (AMD64)]                                                                                                                 
Flavor: Anaconda Python
Executable: ~\Documents\Mosaic\stitching\.pixi\envs\dev\python.EXE
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: ~\AppData\Local\Nuitka\Nuitka\Cache\DOWNLO~1\gcc\x86_64\14.2.0posix-19.1.1-12.0.0-msvcrt-r2\mingw64\bin\gcc.exe (gcc 14.2.0).
```
I installed nuitka with `pixi add --pypi nuitka` and `pixi install`, i.e. latest version (2.5.6) from PyPI.

Python is installed by pixi, i.e. Anaconda Python 3.11 is used.

I am in pixi environment, which has `pillow` installed.

Info is from the lock file:

```
- pypi: https://files.pythonhosted.org/packages/dc/83/1470c220a4ff06cd75fc609068f6605e567ea51df70557555c2ab6516b2c/pillow-11.0.0-cp311-cp311-win_amd64.whl
  name: pillow
  version: 11.0.0
  sha256: 52a2d8323a465f84faaba5236567d212c3668f2ab53e1c74c15583cf507a0291
  requires_dist:
  - furo ; extra == 'docs'
  - olefile ; extra == 'docs'
  - sphinx>=8.1 ; extra == 'docs'
  - sphinx-copybutton ; extra == 'docs'
  - sphinx-inline-tabs ; extra == 'docs'
  - sphinxext-opengraph ; extra == 'docs'
  - olefile ; extra == 'fpx'
  - olefile ; extra == 'mic'
  - check-manifest ; extra == 'tests'
  - coverage ; extra == 'tests'
  - defusedxml ; extra == 'tests'
  - markdown2 ; extra == 'tests'
  - olefile ; extra == 'tests'
  - packaging ; extra == 'tests'
  - pyroma ; extra == 'tests'
  - pytest ; extra == 'tests'
  - pytest-cov ; extra == 'tests'
  - pytest-timeout ; extra == 'tests'
  - typing-extensions ; python_full_version < '3.10' and extra == 'typing'
  - defusedxml ; extra == 'xmp'
  requires_python: '>=3.9'
```

```
- pypi: https://files.pythonhosted.org/packages/05/50/436fe8b1f8776ca0f58b8f7ef5a1817665beca1fa6e6b6c2456015d4d0a8/Nuitka-2.5.6.tar.gz
  name: nuitka
  version: 2.5.6
  sha256: 14b859ae95a12021d72e016365354578bd73d3d4aa4430f1dd54d649352a1f79
  requires_dist:
  - ordered-set>=4.1.0
  - zstandard>=0.15
```

Not sure if anything else is relevant here. Can share the full lock file if needed.

Step-by-step example:

1. `pixi init --format pyproject`. Here, pixi will create also src/pkg file structure.

2. Set `requires-python = "== 3.11"` instead of some other `requires-python` in the `pyproject.toml`.

3. `pixi add --pypi pillow`

4. `pixi add --pypi nuitka` 

5. Create `my_pkg_file.py` in src/pkg folder. Add `from PIL import Image`.

6. Create `run.py` in the project folder (i.e. on same level where src is). Add a single line `from test_pillow import my_pkg_file`.

7. `pixi run python -m nuitka --standalone .\run.py`. This runs successfully for me.
8. Now just run the resulting `run.exe` in `run.dist` folder:
```
ImportError: LoadLibraryExW 'C:\Users\Admin\DOCUME~1\PC\TEST_P~1\RUN~1.DIS\PIL\_imaging.pyd' failed: The specified module could not be found.
```
As me, you should probably see the "missing" `_imaging.pyd` actually present in the `run.dist` folder.

Note that Nuitka compiled fined & exe run fine for me with the setup mentioned above (with real project including opencv, numpy, pandas, scipy and other more complicated packages) and only addition of pillow/PIL broke it. I isolated this test example to contain only PIL to make sure that it is indeed the source of it.

The interpreted version (`pixi run python run.py`) does not have any import issues or version-check errors.

**Comment by kayhayen** [comment]
Did you try your example without using `pixi` ?

**Comment by iegorval** [comment]
> Did you try your example without using `pixi` ?
 I had time to test that today, and that does not happen with a regular Python environment + `pip`, while `pillow` and `nuitka` version are kept same. 

I am curious whether it might be caused by Anaconda vs CPython? If you think that makes sense, I can also install conda to test it in a conda environment.

**Comment by iegorval** [comment]
I have now also managed to try with Nuitka 2.4.11 and this error does not happen there: neither for the toy example nor for the actual project.

**Comment by kayhayen** [comment]
Ok, so this should be bisectable then, thanks for your input. With compilation reports, it would be even better, as we could try and diff those to see what went missing, but I can take it from there.

**Comment by kayhayen** [comment]
I managed to reproduce this, indeed with the conda package, on Linux with 3.11 and currently I am bisecting it, but it's a regression of 2.5 somewhere.

**Comment by kayhayen** [comment]
So, this one is guilty of breaking Anaconda.

```
Author: Kay Hayen <kay.hayen@gmail.com>
Date:   Sun Oct 27 08:16:17 2024 +0100

    Windows: Avoid even scanning in PATH if we don't want DLLs 

    * This should avoid encoding issues with some paths that   
      won't work for short filenames encoding to latin1.
```

What's going on is that in `%PATH%` in an environment of conda, there will be directories that contain DLLs to use. I initially thought that I saw it on Linux too, but I must have been confused. Happens only on Windows, where DLLs are loaded from it.

The solution is of course simple, we just make sure to keep `%PATH%` entries that are pointing to inside the conda prefixes, and then it ought to work.

For my MingGW64 bash this resolves as this with the environment I used to bisect this.

```
/c/Users/kayha/anaconda3/envs/pil-test:/c/Users/kayha/anaconda3/envs/pil-test/Library/mingw-w64/bin:/c/Users/kayha/anaconda3/envs/pil-test/Library/usr/bin:/c/Users/kayha/anaconda3/envs/pil-test/Library/bin:/c/Users/kayha/anaconda3/envs/pil-test/Scripts:/c/Users/kayha/anaconda3/envs/pil-test/bin:/c/Users/kayha/anaconda3/condabin:
```

**Comment by kayhayen** [comment]
Thanks for your report. This is fixed on the factory branch, which is a development version that is under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

**Comment by kayhayen** [comment]
This is fixed in 2.5.8 which saw a mini sprint, where I added Anaconda coverage on Nuitka-Watch, largely for Windows, but also with opencv (and hence numpy) on all 3 platforms. I saw that a generic warning is given about unnecessary configuration for numpy, that will be removed at a later date.

This should prevent Anaconda from regressing again in the future.
